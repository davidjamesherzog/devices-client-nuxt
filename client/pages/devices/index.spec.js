import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';
import Buefy from 'buefy';
import Devices from './index.vue';

describe('Devices', () => {
  describe('fetch', () => {
    test('should return networks', async (done) => {
      const page = new Devices();
      const mockContext = {
        list: jest.fn()
      };
      await page.$options.fetch.bind(mockContext)();
      expect(mockContext.list).toHaveBeenCalled();
      done();
    });
  });

  describe('template', () => {
    let localVue;
    let localStore;

    beforeEach(() => {
      localVue = createLocalVue();
      localVue.use(Buefy);
      localVue.use(Vuex);
      localStore = new Vuex.Store({
        modules: {
          device: {
            namespaced: true,
            state: {
              _devices: []
            }
          }
        }
      });
    });

    test('is a Vue instance', () => {
      const wrapper = shallowMount(Devices, {
        localVue,
        mocks: { $store: localStore }
      });
      const title = wrapper.find('[data-device-title]');
      expect(title.text()).toEqual('Devices');
      const list = wrapper.find('[data-device-list]');
      expect(list).toBeDefined();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
