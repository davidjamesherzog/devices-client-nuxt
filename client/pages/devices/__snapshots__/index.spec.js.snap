// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Devices template is a Vue instance 1`] = `
<section class="section">
  <h2 data-device-title="" class="title is-3 has-text-grey">Devices</h2>
  <device-list-stub devices="" data-device-list=""></device-list-stub>
</section>
`;
